code/datums/elements/beauty.dm

```diff
@@ -0,0 +1,69 @@
+/**
+ * Beauty element; makes the indoor area the parent is in prettier or uglier depending on the beauty var value.
+ * Clean and well decorated areas lead to positive moodlets for passerbies;
+ * Shabbier, dirtier ones lead to negative moodlets EXCLUSIVE to characters with the snob quirk.
+ */
+/datum/element/beauty
+	element_flags = ELEMENT_BESPOKE|ELEMENT_DETACH
+	id_arg_index = 2
+	var/beauty = 0
+	/**
+	  * Assoc list of atoms as keys and number of time the same element instance has been attached to them as assoc value.
+	  * So things don't get odd with same-valued yet dissimilar beauty modifiers being added to the same atom.
+	  */
+	var/beauty_counter = list()
+
+/datum/element/beauty/Attach(datum/target, _beauty)
+	. = ..()
+	if(!isatom(target) || isarea(target))
+		return ELEMENT_INCOMPATIBLE
+
+	beauty = _beauty
+
+	if(!beauty_counter[target] && ismovable(target))
+		var/atom/movable/mov_target = target
+		mov_target.become_area_sensitive(BEAUTY_ELEMENT_TRAIT)
+		RegisterSignal(mov_target, COMSIG_ENTER_AREA, .proc/enter_area)
+		RegisterSignal(mov_target, COMSIG_EXIT_AREA, .proc/exit_area)
+
+	beauty_counter[target]++
+
+	var/area/A = get_area(target)
+	if(A && !A.outdoors)
+		A.totalbeauty += beauty
+		A.update_beauty()
+
+/datum/element/beauty/proc/enter_area(datum/source, area/A)
+	SIGNAL_HANDLER
+
+	if(A.outdoors)
+		return
+	A.totalbeauty += beauty * beauty_counter[source]
+	A.update_beauty()
+
+/datum/element/beauty/proc/exit_area(datum/source, area/A)
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names